client/src/Components/QuestionsContainer/QuestionsList/QuestionsListEntry/AnswerModal/AnswerModal.jsx

```diff
@@ -31,91 +32,105 @@ const AnswerModal = (props) => {
-    let fileData = new FileReader()
-    fileData.onloadend = handleFile
-    fileData.readAsDataURL(file)
-  }
-
-  const validateEmail = (mail) => {
-    // eslint-disable-next-line no-console
-    if (/^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/.test(mail)) {
-      // eslint-disable-line no-eval
-      return true
-    }
-    return false
-  }
-
-  const validateBlank = () => {
-    if (validateEmail(email)) {
-      setIsValidEmail(true)
-    }
-    if (nickname !== "") {
-      setIsValidNickname(true)
-    }
-    if (answerBody !== "") {
-      setIsValidAnswerBody(true)
-    }
+  const resetAnswerModal = () => {
+    setEmail("")
+    setNickname("")
+    setAnswerBody("")
+    setPhotoPreviews([])
+    setFiles([])
+    setPhotos([])
+    setValidated(false)
   }
 
-  const handleSubmit = () => {
+  const handleSubmit = (event) => {
+    setValidated(true)
+    event.preventDefault()
     // post if all sections are valid
-    if (isValidEmail && isValidNickname && isValidAnswerBody) {
+    const form = event.currentTarget
+    if (form.checkValidity() === true) {
+      // console.log(form.checkValidity())
       context.postAnswer(
         props.questionId,
         answerBody,
         nickname,
         email,
-        [photos],
+        photos,
         (response) => {
           setShow(false)
+          props.getAnswersData()
         }
       )
+    }
+  }
+
+  const ImagePreviews = (props) => {
+    if (props.url) {
+      return <img src={props.url} width="100" height="100" />
     } else {
-      const invalidFields = []
-
-      if (!isValidEmail) {
-        invalidFields.push("Email")
-      }
-      if (!isValidNickname) {
-        invalidFields.push(" Nickname")
-      }
-      if (!isValidAnswerBody) {
-        invalidFields.push(" Answer Body")
-      }
-
-      alert(
-        "You must enter the following:\n" +
-          invalidFields.map((invalidField) => invalidField)
+      return null
+    }
+  }
+
+  const InputImages = () => {
+    if (photoPreviews.length < 5) {
+      return (
+        <Form.Control
+          type="file"
+          onChange={(e) => {
+            // console.log(e.target.files)
+            setPhotoPreviews([
+              ...photoPreviews,
+              URL.createObjectURL(e.target.files[0]),
+            ])
+            setFiles([...files, e.target.files[0]])
+          }}
+        />
       )
+    } else {
+      return null
     }
   }
 
+  const handleFileUpload = (e) => {
+    e.preventDefault()
+    const urls = []
+    const uploaders = files.map((file) => {
+      const formData = new FormData()
+      formData.append("file", file)
+      // formData.append("tags", `codeinfuse, medium, gist`)
+      formData.append("upload_preset", "wcjmqzvr")
+      // formData.append("api_key", "353977359896185")
+      // formData.append("timestamp", (Date.now() / 1000) | 0)
+
+      axios
+        .post("https://api.cloudinary.com/v1_1/dqab317k6/image/upload", formData)
```

Comment:
Looks like this is used to create a URL that can be stored/used later? I might use this too if it works!

Reply:
It works great! You can use the same format to upload images if you'd like. I created an account on cloudinary for it so if you want you could just use that same endpoint.